src/Controls/samples/Controls.Sample/Startup.cs

```diff
@@ -0,0 +1,64 @@
+﻿using System;
+using System.Collections.Generic;
+using Maui.Controls.Sample.Pages;
+using Maui.Controls.Sample.Services;
+using Maui.Controls.Sample.ViewModel;
+using Microsoft.Extensions.Configuration;
+using Microsoft.Extensions.DependencyInjection;
+using Microsoft.Extensions.DependencyInjection.Extensions;
+using Microsoft.Maui;
+using Microsoft.Maui.Controls.Compatibility;
+using Microsoft.Maui.Hosting;
+
+namespace Maui.Controls.Sample
+{
+	public class Startup : IStartup
+	{
+		public readonly static bool UseXamlPage = false;
+
+		public void Configure(IAppHostBuilder appBuilder)
+		{
+			appBuilder
+				.RegisterCompatibilityRenderers()
+				.UseMauiApp<MyApp>()
+				.ConfigureAppConfiguration((hostingContext, config) =>
+				{
+					config.AddInMemoryCollection(new Dictionary<string, string>
+					{
+						{"MyKey", "Dictionary MyKey Value"},
+						{":Title", "Dictionary_Title"},
+						{"Position:Name", "Dictionary_Name" },
+						{"Logging:LogLevel:Default", "Warning"}
+					});
+				})
+				.UseMauiServiceProviderFactory(true)
+				//.UseServiceProviderFactory(new DIExtensionsServiceProviderFactory())
```

Comment:
By default you don't need this line, but then you don't get the constructor injection. The first line adds the internal, fast, but basic, injection. Using a full DI system can replace this line using the second line.